File: backend/src/main/java/com/inso/sila/service/MembershipService.java
package com.inso.sila.service;

import com.inso.sila.endpoint.dto.studio.membership.MembershipDto;

import java.util.List;

public interface MembershipService {

    MembershipDto addMembership(MembershipDto membershipDto);

    MembershipDto deleteMembership(Long id);

    List<MembershipDto> getMembershipsForUser();

    boolean hasMembershipForStudio(Long studioId);
}

File: frontend/src/app/services/membership.service.ts
import { Injectable } from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Membership} from "../dtos/membership";

const baseUri = environment.backendUrl + '/api/v1/memberships';

@Injectable({
  providedIn: 'root'
})
export class MembershipService {

  constructor(private httpClient: HttpClient,) { }


  getMembershipsForUser(): Observable<Membership[]> {
    return this.httpClient.get<Membership[]>(`${baseUri}`);
  }


  addMembership(membershipDto: Membership): Observable<Membership> {
    return this.httpClient.post<Membership>(`${baseUri}`, membershipDto);
  }


  deleteMembership(membershipId: number): Observable<void> {
    return this.httpClient.delete<void>(`${baseUri}/${membershipId}`);
  }

  hasMembershipForStudio(studioId: number): Observable<boolean> {
    return this.httpClient.get<boolean>(`${baseUri}/hasMembershipForStudio/` + studioId);
  }
}
